fix: tmux `allow-passthrough` is off by default

gszr opened the pull request:
As of tmux 3.3a, the `allow-passthrough` option is off by default. Further, as mentioned in https://github.com/tmux/tmux/wiki/FAQ:
```
The passthrough escape sequence is no longer necessary for changing the
cursor colour or style as tmux now has its own support (see the Cs, Cr,
Ss and Se capabilities).
```

This commit checks whether the `allow-passthrough` is set, and uses it; otherwise, no wrapping is performed.

cbaconnier commented on bin/dynamic-colors:
```diff
@@ -49,7 +49,11 @@ send_escape_sequence () {
-  [ ! -z "$TMUX" ] && escape_sequence="${DSC}tmux;${ESC}${escape_sequence}${ESC}\\"
+  tmux_allow_passthrough=`tmux show-options -gw 2>/dev/null | awk '$1 == "allow-passthrough" { print $2}'`
+  if [ $tmux_allow_passthrough == on ]; then
+    echo 'here'
```
Woops :grin:

gszr replied:
Woops! Fixed :)

cbaconnier commented:
Thanks ! I tried locally and it fixed my issue.

It would also close https://github.com/sos4nt/dynamic-colors/issues/23

gszr committed:
fix: tmux `allow-passthrough` is off by default

As of tmux 3.3a, the `allow-passthrough` option is off by default.
Further, as mentioned in https://github.com/tmux/tmux/wiki/FAQ:
```
The passthrough escape sequence is no longer necessary for changing the
cursor colour or style as tmux now has its own support (see the Cs, Cr,
Ss and Se capabilities).
```

This commit checks whether the `allow-passthrough` is set, and uses it;
otherwise, no wrapping is performed.
